rutil/Data.hxx

```diff
@@ -1126,6 +1148,35 @@ bool operator<(const Data& lhs, const Data& rhs);
 bool operator<(const Data& lhs, const char* rhs);
 bool operator<(const char* lhs, const Data& rhs);
 
+
+// Overload operator== for resip::Data and std::string
+bool operator==(const resip::Data& lhs, const std::string& rhs)
```

Comment:
Please review the implementation to avoid the compilation error:
```
/__w/resiprocate/resiprocate/rutil/Data.hxx:1154: multiple definition of `resip::operator==(resip::Data const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)';
```
It can be checked here - https://github.com/resiprocate/resiprocate/actions/runs/11493692672/job/31989841140?pr=382

Reply:
Thanks, fixed.